src/PrestaShopBundle/Form/Extension/NumberTypeExtension.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShopBundle\Form\Extension;
+
+use PrestaShop\PrestaShop\Core\Localization\LocaleInterface;
+use Symfony\Component\Form\AbstractTypeExtension;
+use Symfony\Component\Form\Extension\Core\DataTransformer\NumberToLocalizedStringTransformer;
+use Symfony\Component\Form\Extension\Core\Type\NumberType;
+use Symfony\Component\Form\FormBuilderInterface;
+
+class NumberTypeExtension extends AbstractTypeExtension
+{
+    /**
+     * @var LocaleInterface
+     */
+    private $locale;
+
+    /**
+     * @param LocaleInterface $locale
+     */
+    public function __construct(
+        LocaleInterface $locale
+    ) {
+        $this->locale = $locale;
+    }
+
+    public static function getExtendedTypes(): iterable
+    {
+        return [NumberType::class];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function buildForm(FormBuilderInterface $builder, array $options)
+    {
+        // TODO: in 9.0, add getLocaleForNumbers to LocaleInterface and remove this check
+        if (method_exists($this->locale, 'getLocaleForNumbers')) {
+            $viewTransformers = $builder->getViewTransformers();
```

Comment:
Nice this way, but it probably deserves a little comment explaining why we do this likt
```suggestion
            // We only want to replace/adapt the NumberToLocalizedStringTransformer so we backup the current transformers so we can add the other ones
            $viewTransformers = $builder->getViewTransformers();
```